src/pages/order/payment.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {FC, useContext, useState} from 'react';
+import {Alert} from 'antd';
+import {Context, ContextProps} from '../../App';
+import { withRouter } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+
+interface PaymentPageProps {
+	history: any;
+}
+
+export type paymentDetails = {
+	name?: string,
+	address?: string,
+	city?: string,
+	email?: string,
+	phone?: string,
+	cardNumber?: number,
+	expiryDate?: string,
+	cvc?: number,
+}
+
+const PaymentPage: FC<PaymentPageProps> = ({history}) => {
+	const { register, handleSubmit } = useForm();
+
+	const context = useContext<ContextProps>(Context);
+	console.log('context', context);
+	const [paymentError, setPaymentError] = useState(false);
+
+
+	const pay = async (details: paymentDetails): Promise<void>=> {
+		const paymentValid = details.email && details.name && details.address && details.phone && details.city && details.cardNumber && details.cvc && details.expiryDate;
```

Comment:
instead of this logic, please implement proper validation. react hooks form has a good validation system using yup.